config/routes.rb

```diff
@@ -3,4 +3,13 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  resources :dishes, only: [:show] do
+    resources :ingredients, only: [:create], controller: "dish_ingredients"
+  end
+
+  resources :chefs, only: [:show,] do 
+    resources :ingredients, only: [:index]
```

Comment:
Great RESTful routing! Another option here could be to use a nested controller since the ingredients are nested under chefs (you could add `controller: "chefs/ingredients"` at the end of line 12 to force the controller)